<!DOCTYPE html>
<!--
Copyright (c) 2013 Samsung Electronics Co., Ltd.

Licensed under the Apache License, Version 2.0 (the License);
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


Authors:
        Beata Koziarek <b.koziarek@samsung.com>

-->
<html>
<head>
<title>File_openStream_onerror_TypeMismatch</title>
<meta charset="utf-8"/>
<script type="text/javascript" src="support/unitcommon.js"></script>
</head>

<body>
<div id="log"></div>
<script>
//==== TEST: File_openStream_onerror_TypeMismatch
//==== LABEL Check if File::openStream() method throws exception for wrong type of onerror
//==== PRIORITY: P2
//==== SPEC: Tizen Web API:IO:Filesystem:File:openStream M
//==== SPEC_URL https://developer.tizen.org/help/topic/org.tizen.web.device.apireference/tizen/filesystem.html
//==== TEST_CRITERIA MC

var t = async_test(document.title),
    resolveSuccess, resolveError, conversionTable, openStreamError, i, openStreamSuccess,
    exceptionName = "TypeMismatchError";
t.step(function () {
    openStreamSuccess = t.step_func(function (fs) {
        fs.close();
        assert_unreached("openStream() success callback invoked: should not be called");
    });
    resolveSuccess = t.step_func(function (dir) {
        conversionTable = getTypeConversionExceptions("functionObject", true);
        for (i = 0; i < conversionTable.length; i++) {
            openStreamError = conversionTable[i][0];
            exceptionName = conversionTable[i][1];

            assert_throws({name: exceptionName},
                function () {
                    dir.openStream("r", openStreamSuccess, openStreamError);
                }, exceptionName + " should be thrown - given incorrect errorCallback.");
        }
        t.done();
    });

    resolveError = t.step_func(function (error) {
        assert_unreached("resolve() error callback invoked: name:" + error.name + "msg:" + error.message);
    });

    tizen.filesystem.resolve("documents", resolveSuccess, resolveError, "rw");
});

</script>
</body>
</html>
